### ISSUE TITLE
how to train teacher model

### ISSUE THREAD

**Issue by shaoyandea** [skyshoumeng/Label_Assisted_Distillation#2]
your work is great, i am very instereted in this paper , but i found there is not training code and preprained model of teacher model, can you release it?

**Comment by skyshoumeng** [comment]
Sorry for the late reply. During training, the teacher model only needs to shuffle the labels and concatenate them with the RGB image, forming a 4-channel input. An example code is as follows:

batch_size = 4
number_classes = 20
// assume label: [batch_size, 1 H , W],  img: [batch_size, 3 H , W]
H = label.shape[2]
W = label.shape[3]

label = label.float()
for j in range(batch_size):
    for k in range(number_classes):
        // replace label id == k to torch.randn(1)
        label[j,...][torch.abs(label[j,...]-k)<1e-5] = torch.randn(1).cuda()

// add pixel wise noise
noised_label = label + torch.randn(batch_size,1,H,W).cuda() * pixel_wise_nosie ## 0.1 or 0.01
img_enhanced = torch.cat([img, noised_label], dim=1)

ouput = model(img_enhanced)
...
...
..

If you have any other questions, feel free to contact me for discussion.

**Comment by shaoyandea** [comment]
thank your for your reply, are teacher network and student network trained and optimized at the same time?  Is the total loss during training the sum of formulas 7 and 8 in the paper  @skyshoumeng

**Comment by skyshoumeng** [comment]
In the method, the teacher model and student model are trained separately. Formula 7 in the paper represents the loss function for the teacher model training, while Formula 8 represents the loss function for the student model training. If you have any further questions, feel free to discuss them with me.
